### ISSUE TITLE
appindicator not behaving properly with topbar moved to bottom of screen

### ISSUE THREAD

**Issue by fishears** [ubuntu/yaru#641]
Not sure if you'll dismiss this as non-standard desktop config but many users have their top bar at the bottom with extensions like dashtopanel...

The appindicator for steam works with the topbar at the top i.e. the appindicator menu drops down from the icon but if I switch my topbar to the bottom of the screen the appindicator menu does not appear at all. The only other appindicator I have is Insync and that works fine at the top or bottom of the screen. 

The steam appindicator works fine at the top and bottom with the default ubuntu shel theme

(Ubuntu 18.04)

**Comment by Feichtmeier** [comment]
> but **many** users have their top bar at the bottom with extensions like dashtopanel...

I highly doubt that.
Anyways... Could you be sure that this behaviour only happens with the communitheme snap?

**Comment by fishears** [comment]
@Feichtmeier consider this explicit confirmation that this issue relates soley to the communitheme snap on my setup

**Comment by clobrano** [comment]
I was about to say I couldn't reproduce your isse (see picture below), however I am on development environment and not on snap, so I can't say that with 100% certainty

![screenshot from 2018-07-21 17-12-38](https://user-images.githubusercontent.com/2883614/43037086-51167cd2-8d09-11e8-9dff-5da05a6e86f3.png)

**Comment by fishears** [comment]
I have the issue with the snap and the PPA build. It was working fine until recently. The standard ubuntu.css works all the time though. Wish I knew where to look for the crucial difference ;)

**Comment by Feichtmeier** [comment]
You have the snap installed and the ppa installed?

**Comment by fishears** [comment]
No. I have tested both individually.

**Comment by clobrano** [comment]
Unfortunately I can't test steam on a VM, which is the only way I can test the snap installation

If you installed steam some time ago, I suggest you to reinstall the official package, they might have integrated the patch @didrocks provided in electron

**Comment by fishears** [comment]
Thanks but the steam client updates itself automatically as builds are released. By way of new information, I can offer this: If I disable "Ubuntu Appindicator" extension and instead use "Topicons Plus" extension to do the same job, I get a menu for the Steam indicator (albeit an unthemed menu). Does this suggest a problem between Communitheme and Ubuntu Appindictor? I say this because other shell themes behave properly...

**Comment by clobrano** [comment]
It might be, thank you

**Comment by fishears** [comment]
I see this still has the "info-needed" label. Is there anything more you need me to provide?

**Comment by clobrano** [comment]
@fishears no, thank you, I am going to remove the label

**Comment by fishears** [comment]
I've been running the standard Ubuntu desktop layout for a few days (Panel at the top + Ubuntu Dock on the left). The Steam appindicator menu works as expected with Yaru but I noticed just how much of my screen my Steam menu takes up when active - almost to the bottom. From this I decided to try again with Dash to Panel at the bottom of the screen but with a smaller icon size (making it similar size to the standard Ubuntu panel. The Steam menu now appears correctly. 
I take it that the appindicator menu simply does not display if there is insufficient space above (or below) the panel. I therefore doubt this is an issue for the Yaru team - more for the ubuntu appindicator team. Maybe this can be closed.
